### ISSUE TITLE
v3.0.1 polyfills still not working properly in latest release

### ISSUE THREAD

**Issue by niftylettuce** [amilajack/eslint-plugin-compat#197]
With contents: `{ settings: { polyfills: [ 'Promise' ] } }`, the error message is:

```sh
Promise.resolve() is not supported in Opera Mobile 12.1, Opera Mini all, IE Mobile 10, IE 10, Blackberry Browser 7  compat/compat

Promise.reject() is not supported in Opera Mobile 12.1, Opera Mini all, IE Mobile 10, IE 10, Blackberry Browser 7   compat/compat
```

With contents: `{ settings: { polyfills: [ 'promises' ] } }`, the error message is:

```sh
Promise is not supported in Opera Mobile 12.1, Opera Mini all, IE Mobile 10, IE 10, Blackberry Browser 7            compat/compat

Promise.resolve() is not supported in Opera Mobile 12.1, Opera Mini all, IE Mobile 10, IE 10, Blackberry Browser 7  compat/compat

Promise.reject() is not supported in Opera Mobile 12.1, Opera Mini all, IE Mobile 10, IE 10, Blackberry Browser 7   compat/compat
```

Which leads me to have to write out all the methods in order to pass (which I don't think is working as you intended, since `Promise` alone should work just fine, no?).

With contents: `{ settings: { polyfills: [ 'Promise', 'Promise.resolve', 'Promise.reject' ] } }`, the error message is empty.

Also, in v2 to v3, you cannot use `promises` anymore, you have to use `Promise`.  It would be nice to be able to use the generic caniuse ID again as we could in v2.

P.S. As a side note, could you please use `np` to publish new versions?  The releases with this package have not been consistent at all and it should help.   See <https://github.com/sindresorhus/np/> for more info.

--- 

Btw I became your first supporter on Patreon just now! 🎉 

Hope this helps 🙌 🙇

**Comment by amilajack** [comment]
Thank you! You're awesome 🎉 🎉 🎉 . The `@next` branch has the fix for this. Waiting for people to try it out before I push the patch for `v3.0.2`. Let me know if that it works for you.

**Comment by niftylettuce** [comment]
Oh, I had just filed #191 and didn't assume you used `np`.  Will close that issue now that I know you do for the future.

**Comment by amilajack** [comment]
Btw I added a test case for the bug you're filing: https://github.com/amilajack/eslint-plugin-compat/pull/198. This passes in master and in `eslint-plugin-compat@next`. If `@next` does not work for you can you please PR a failing test case. Thanks!

**Comment by niftylettuce** [comment]
I don't use `@next` versions in my projects in order to maintain stability.  But I did test it and it seemed to work OK.
